### PR TITLE
drivers: Various minor improvments of the cc110x driver

### PR DESCRIPTION
**Pull request by maribu**
This PR contains a number of minor improvements of the cc110x driver:
- Improved code readability by fixing documentation, replacing magic numbers by preprocessor macros, and fixing indentation
- Improved debug output `ENABLE_DEBUG` is set
- Implemented a missing function that is referenced if `ENABLE_DEBUG` is set

This PR fixes issue https://github.com/RIOT-OS/RIOT/issues/8115 at least partly.

Even though this PR is completely boring, I tested the code on the MSBA2 boards, on which the cc110x works just like before.

### CODE REVIEW

File: drivers/cc110x/include/cc110x-defines.h [hunk 1]
```diff
@@ -208,6 +208,35 @@ extern "C" {
 #define CC110X_RXFIFO       (0x3F)      /**< RX FIFO: Read operations read from the RX FIFO (SB: +0x80; BURST: +0xC0) */
 /** @} */
 
+
+/**
+ * @name GDO configuration values
+ * @brief Values that can be written to the GDO0, GDO1 and GDO2 configuration
```
Comment (smlng) [inline drivers/cc110x/include/cc110x-defines.h]: omit the `@brief` here and add an empty line between `@name`

File: drivers/cc110x/include/cc110x-defines.h [hunk 2]
```diff
@@ -208,6 +208,35 @@ extern "C" {
...
+ *        registers
+ * @{
+ */
+
+/// GDO goes high when RX FIFO is filled at or above threshold
```
Comment (smlng) [inline drivers/cc110x/include/cc110x-defines.h]: to document each define we either use 
```C
/**
 * @brief GDO goes high when RX FIFO is filled at or above threshold
 */
#define CC110X_GDO_HIGH_ON_RX_FIFO_ABOVE_THRESHOLD   (0x00)
```

or

```C
#define CC110X_GDO_HIGH_ON_RX_FIFO_ABOVE_THRESHOLD   (0x00) /**< GDO goes high when RX FIFO is filled at or above threshold */
```

File: drivers/cc110x/include/cc110x-defines.h [hunk 3]
```diff
@@ -208,6 +208,35 @@ extern "C" {
...
+#define CC110X_GDO_HIGH_ON_RX_FIFO_OVERFLOW          (0x04)
+/// GDO goes high when TX FIFO underflows
+#define CC110X_GDO_HIGH_ON_TX_FIFO_UNDERFLOW         (0x05)
+/// GDO goes high when sync word was just received until the packet is
+/// fully received, or when sync word has been send until packet is fully send
+#define CC110X_GDO_HIGH_ON_SYNC_WORD                 (0x06)
+/// GDO goes high when a packet is received and CRC is correct. Goes back to
+/// low when first byte of RX fifo has been read
+#define CC110X_GDO_HIGH_ON_PACKET_RECEIVED           (0x07)
+/// @}
```
Comment (smlng) [inline drivers/cc110x/include/cc110x-defines.h]: should be `/** @} */`

File: drivers/include/cc110x.h [hunk 4]
```diff
@@ -33,11 +33,11 @@ extern "C" {
  * @brief   Struct for holding cc110x IO parameters
  */
 typedef struct cc110x_params {
-    spi_t spi;          /**< what */
-    gpio_t cs;          /**< does */
-    gpio_t gdo0;        /**< this */
-    gpio_t gdo1;        /**< look */
-    gpio_t gdo2;        /**< like */
+    spi_t spi;          ///< SPI bus the CC110x is connected to
```
Comment (smlng) [inline drivers/include/cc110x.h]: please use comment style as before and explained above, not `///`

File: drivers/cc110x/cc110x-rxtx.c [hunk 5]
```diff
@@ -133,14 +154,17 @@ static void _rx_read_data(cc110x_t *dev, void(*callback)(void*), void*arg)
         else {
             DEBUG("%s:%s:%u crc-error\n", RIOT_FILE_RELATIVE, __func__, __LINE__);
             dev->cc110x_statistic.packets_in_crc_fail++;
+#if defined(MODULE_OD) && ENABLE_DEBUG
+            od_hex_dump(pkt_buf->packet.data, pkt_buf->packet.length-3,
```
Comment (smlng) [inline drivers/cc110x/cc110x-rxtx.c]: minor nit, please use spaces in math expressions, i.e., `(pkt_buf->packet.length - 3)`

File: drivers/cc110x/cc110x-rxtx.c [hunk 6]
```diff
@@ -37,12 +37,32 @@
...
 #include "cpu_conf.h"
 #include "cpu.h"
+#include "od.h"
 
 #include "log.h"
 
 #define ENABLE_DEBUG (0)
 #include "debug.h"
 
+const char *cc110x_state_to_text(uint8_t state)
```
Comment (smlng) [inline drivers/cc110x/cc110x-rxtx.c]: nice catch, there wasn't even an implementation for that before 😮

**Comment by maribu** [comment]
@smlng: Thanks for the review. I believe I have now addressed all issues you pointed out.

**Comment by smlng** [comment]
> just need to be patient

kind of, at least to give others some time to look it over, too.

File: drivers/cc110x/cc110x-rxtx.c [hunk 7]
```diff
@@ -37,12 +37,32 @@
 
 #include "cpu_conf.h"
 #include "cpu.h"
+#include "od.h"
```
[the remark this answers is not in the document]
Comment (maribu) [reply drivers/cc110x/cc110x-rxtx.c]: Done :-)

**Comment by aabadie** [comment]
All green here, merging this one, thanks @maribu !

**Pull request merged**